cli/pack/docker.go

```diff
@@ -0,0 +1,110 @@
+package pack
+
+import (
+	_ "embed"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/otiai10/copy"
+	"github.com/tarantool/tt/cli/cmdcontext"
+	"github.com/tarantool/tt/cli/config"
+	"github.com/tarantool/tt/cli/docker"
+	"github.com/tarantool/tt/cli/templates/engines"
+)
+
+//go:embed templates/Dockerfile.pack.build
+var buildDockerfile []byte
+
+// RunInDocker runs tt pack in docker container.
+func RunInDocker(cmdCtx *cmdcontext.CmdCtx, opts config.CliOpts, cmdArgs []string) error {
+	tmpDir, err := ioutil.TempDir("", "docker_pack_ctx")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tmpDir)
+
+	envDir := filepath.Dir(cmdCtx.Cli.ConfigPath)
+
+	goTextEngine := engines.NewDefaultEngine()
+	dockerfileText, err := goTextEngine.RenderText(string(buildDockerfile),
+		map[string]string{
+			"tnt_version": cmdCtx.Cli.TarantoolVersion,
+			"env_dir":     filepath.Base(envDir),
+		})
+	if err != nil {
+		return err
+	}
+
+	// Write docker file (rw-rw-r-- permissions).
+	if err = ioutil.WriteFile(filepath.Join(tmpDir, "Dockerfile"),
+		[]byte(dockerfileText),
+		0664); err != nil {
+		return err
+	}
+
+	// Remove --use-docker from args.
+	for i, arg := range cmdArgs {
+		if arg == "--use-docker" {
+			cmdArgs = append(cmdArgs[:i], cmdArgs[i+1:]...)
```

Comment:
Did you check how it works if multiple --use-docker options provided?

Reply:
Yes, it works correct.